src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py

```diff
@@ -0,0 +1,314 @@
+    if not _single(collection):
+        raise CLIError("usage error: [--filters FILTER | --my-vms | --claimable]")
+
+    if namespace.my_vms:
+        # Find out owner object id
+        from azure.cli.core._profile import Profile, CLOUD
+        profile = Profile()
+        cred, _, tenant_id = profile.get_login_credentials(
+            resource=CLOUD.endpoints.active_directory_graph_resource_id)
+        graph_client = GraphRbacManagementClient(cred,
+                                                 tenant_id,
+                                                 base_url=CLOUD.endpoints.active_directory_graph_resource_id)
+        subscription = profile.get_subscription()
+        object_id = namespace.object_id or _get_object_id(graph_client, subscription=subscription)
+        namespace.filters = "Properties/ownerObjectId eq '{}'".format(object_id)
+    elif namespace.claimable:
+        namespace.filters = 'properties/allowClaim'
+    else:
+        namespace.filters = namespace.filters
+
+
+def _validate_location(namespace):
+    """
+    Selects the default location of the lab when location is not provided while creating vm.
+    """
+    if namespace.location is None:
+        lab_operation = devtestlabs_management_client.lab
+        lab = lab_operation.get_resource(namespace.resource_group, namespace.lab_name)
+        namespace.location = lab.location
+
+
+def _validate_expiration_date(namespace):
+    """
+    Validates expiration date if provided.
+    """
+    if namespace.expiration_date:
+        if datetime.datetime.utcnow().date() >= dateutil.parser.parse(namespace.expiration_date).date():
+            raise CLIError("Expiration date '{}' must be in future.".format(namespace.expiration_date))
+
+
+def _validate_network_parameters(namespace):
+    """
+    Selects the DevTest Lab's virtual network and subnet if not provided and updates namespace
+    """
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    vnet_operation = devtestlabs_management_client.virtual_network
+
+    if not namespace.vnet_name:
+        lab_vnets = list(vnet_operation.list(namespace.resource_group, namespace.lab_name, top=1))
+        if not lab_vnets:
+            err = "Unable to find any virtual network in the '{}' lab.".format(namespace.lab_name)
+            raise CLIError(err)
+        else:
+            lab_vnet = lab_vnets[0]
+            namespace.vnet_name = lab_vnet.name
+            namespace.lab_virtual_network_id = lab_vnet.id
+    else:
+        namespace.lab_virtual_network_id = resource_id(subscription=get_subscription_id(),
+                                                       resource_group=namespace.resource_group,
+                                                       namespace='microsoft.devtestlab',
+                                                       type='labs',
+                                                       name=namespace.lab_name,
```

Comment:
Older labs have a different default lab name - dtl + labName + vnet. We may want to just see if they have only one vnet and default to that one.